### PR TITLE
Jeffrey

### PR DESCRIPTION
**Pull request by Darkglade1**
Finished M6.
Everything should be working but other people should check just in case.

### CODE REVIEW

File: app/src/main/java/com/buzzshelter/Controllers/CSVFile.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package com.buzzshelter.Controllers;
```
Comment (dhuynh38) [inline app/src/main/java/com/buzzshelter/Controllers/CSVFile.java]: Only activities are considered controllers
This class has a lot of business logic that can all be moved to the Model class or as a class a new class in the Model

File: app/src/main/java/com/buzzshelter/Controllers/CSVFile.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CSVFile {
+    InputStream inputStream;
```
Comment (dhuynh38) [inline app/src/main/java/com/buzzshelter/Controllers/CSVFile.java]: I would make this private

File: app/src/main/java/com/buzzshelter/Controllers/CSVFile.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+        try {
+            String csvLine;
+            while ((csvLine = reader.readLine()) != null) {
+                String[] row = csvLine.split(";");
+                resultList.add(row);
+            }
+        }
+        catch (IOException ex) {
+            throw new RuntimeException("Error in reading CSV file: "+ex);
```
Comment (dhuynh38) [inline app/src/main/java/com/buzzshelter/Controllers/CSVFile.java]: nice

File: app/src/main/java/com/buzzshelter/Controllers/DetailedShelterActivity.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+        backButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(getBaseContext(), ViewShelterActivity.class);
+                startActivity(intent);
+            }
+        });
+
+        //gets the passed-in shelter, retrieves its info, and displays it in the listview
```
Comment (dhuynh38) [inline app/src/main/java/com/buzzshelter/Controllers/DetailedShelterActivity.java:39]: nice

File: app/src/main/java/com/buzzshelter/Controllers/EnterShelterActivity.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+ */
+
+public class EnterShelterActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_entershelter);
+
+        final EditText shelterName = findViewById(R.id.enterName);
```
Comment (dhuynh38) [inline app/src/main/java/com/buzzshelter/Controllers/EnterShelterActivity.java:25]: just curious. Why are these all final?

File: app/src/main/java/com/buzzshelter/Controllers/WelcomeActivity.java [hunk 6]
```diff
@@ -18,6 +25,27 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_welcome);
 
+        //creates a default user (for convenience's sake)
```
Comment (dhuynh38) [inline app/src/main/java/com/buzzshelter/Controllers/WelcomeActivity.java:27]: like the idea

**Review by dhuynh38 — COMMENTED** [review]
Everything looks good. I tested all of the features and it works. The only main problem I see is the CSVFile class being listed as a controller. The CSVFile does not interact with the View and just does business logic so I would move that under models and have an instance variable inside model give you access to the CSVFile.

File: app/src/main/java/com/buzzshelter/Controllers/EnterShelterActivity.java (hunk 5 above)
Comment (Darkglade1) [reply app/src/main/java/com/buzzshelter/Controllers/EnterShelterActivity.java:25]: For some reason the "final" modifiers were automatically added when I created the variable.
I have no idea why lol.

File: app/src/main/java/com/buzzshelter/Controllers/CSVFile.java (hunk 2 above)
Comment (Darkglade1) [reply app/src/main/java/com/buzzshelter/Controllers/CSVFile.java]: When I try to make it private, Android Studio gives me a warning that says that "private is not allowed here."

File: app/src/main/java/com/buzzshelter/Controllers/CSVFile.java (hunk 1 above)
Comment (Darkglade1) [reply app/src/main/java/com/buzzshelter/Controllers/CSVFile.java]: Fixed

**Pull request merged**